src/idl_parser.cpp

```diff
@@ -1503,14 +1576,17 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
   for (size_t size = struct_def.sortbysize ? sizeof(largest_scalar_t) : 1; size;
        size /= 2) {
     // Go through elements in reverse, since we're building the data backwards.
+    // TODO(derekbailey): this doesn't work when there are Offset64 fields, as
+    // those have to be built first. So this needs to be changed to iterate over
+    // Offset64 then Offset32 fields.
```

Comment:
so what does it do now? should it error?

Reply:
This was mostly for my testing to test a JSON input to binary output before I had the C++ code generation working. I plan on addressing it after the merge, as its only affecting JSON defined data which is obscure.